app/controllers/achievements_controller.rb

```diff
@@ -5,8 +5,14 @@ class AchievementsController < ApplicationController
   def create
     @achievement = current_user.achievements.build(achievement_params)
 
-    if @achievement.save && @achievement.transition_to(:drafted)
-      flash[:notice] = "'#{@achievement.activity.title}' progress has been saved"
+    if @achievement.save
+      if params[:enrol]
+        @achievement.transition_to(:enrolled)
+        flash[:notice] = "'#{@achievement.activity.title}' has been added to your activity list"
+      else
+        flash[:notice] = "'#{@achievement.activity.title}' progress has been saved"
+        @achievement.transition_to(:drafted)
```

Comment:
Can we just swap these two lines round, so the flash is set second like the block above?